### PR TITLE
Add functions to get more series of subgroups

### PR DESCRIPTION
**Pull request by fingolfin**
This is not yet done, I need to complete the docstrings. But since I mentioned `lower_central_series` elsewhere to @lgoettgens I thought I should at least show this here for now.

### CODE REVIEW

**Comment by lgoettgens** [comment]
Thanks!

If #2644 gets merged before this, the im-/exporting between `Oscar` and `Oscar.LieAlgebras` for `lower_central_series` needs to be redone here.

**Comment by fingolfin** [comment]
This is now ready for review. I've also added examples to a bunch of existing docstrings that had none.

File: gap/OscarInterface/gap/misc.gi [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# When a permutation group gets a stabilizer chain,
+# then we can read of its size from that for basically free.
```
Comment (ThomasBreuer) [inline gap/OscarInterface/gap/misc.gi]:
```suggestion
# then we can read off its size from that for basically free.
```

File: src/Groups/sub.jl [hunk 2]
```diff
@@ -199,6 +292,24 @@ among the nontrivial normal subgroups.
 
 Return the list of characteristic subgroups of `G`,
```
Comment (ThomasBreuer) [inline src/Groups/sub.jl:293]:
```suggestion
Return all characteristic subgroups of `G`,
```

File: src/Groups/sub.jl [hunk 3]
```diff
@@ -617,8 +986,15 @@ end
 """
     derived_subgroup(G::GAPGroup)
 
-Return the derived subgroup of `G`, i.e.,
-the subgroup generated by all commutators of `G`.
+Return the derived subgroup `G'` of `G`, i.e.,
+the subgroup generated by all commutators of `G`,
+together with an embedding `G'` into `G`.
```
Comment (ThomasBreuer) [inline src/Groups/sub.jl:991]:
```suggestion
together with an embedding of `G'` into `G`.
```

**Review by ThomasBreuer — APPROVED** [review]
Nice.
(Just a few suggestions.)

As far as I understand, you want to omit the phrase "the vector of" from the description of the return value in those cases where the ordering of the entries does not have a meaning (for example in `normal_subgroups`), but to use "the vector of" for return values describing series.

**Comment by fingolfin** [comment]
> As far as I understand, you want to omit the phrase "the vector of" from the description of the return value in those cases where the ordering of the entries does not have a meaning (for example in `normal_subgroups`), but to use "the vector of" for return values describing series.

Yes, that's roughly the idea. Then for e.g. `subgroups` in the future we maybe can return a special iterator which only computes those groups on the fly. This won't be a vector anymore.

File: src/Groups/sub.jl [hunk 4]
```diff
@@ -199,6 +292,24 @@ among the nontrivial normal subgroups.
...
+ Permutation group of degree 3 and order 2
+ Permutation group of degree 3 and order 2
+ Permutation group of degree 3 and order 3
+ Permutation group of degree 3 and order 6
+
+julia> characteristic_subgroups(quaternion_group(8))
+3-element Vector{PcGroup}:
+ Pc group of order 1
+ Pc group of order 2
+ Pc group of order 8
```
Comment (ThomasBreuer) [inline src/Groups/sub.jl:309]: The result of the doctest says that the groups should appear in decreasing order.
I have checked with several GAP versions that this behaviour is reproducible.
How did the claimed result with increasing group orders arise?

Comment (fingolfin) [reply src/Groups/sub.jl:309]: Apparently the order changes depending on whether the CRIPS package is loaded or not.

**Pull request merged**